src/sqlast/ddl.rs

```diff
@@ -84,3 +79,106 @@ impl ToString for TableConstraint {
         }
     }
 }
+
+/// SQL column definition
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub struct SQLColumnDef {
+    pub name: SQLIdent,
+    pub data_type: SQLType,
+    pub collation: Option<SQLObjectName>,
+    pub constraints: Vec<ColumnConstraint>,
+}
+
+impl ToString for SQLColumnDef {
+    fn to_string(&self) -> String {
+        format!(
+            "{} {}{}",
+            self.name,
+            self.data_type.to_string(),
+            self.constraints
+                .iter()
+                .map(|c| format!(" {}", c.to_string()))
+                .collect::<Vec<_>>()
+                .join("")
+        )
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Hash)]
+pub enum ColumnConstraint {
+    /// `NULL`
+    ///
+    /// The ANSI specification technically allows NULL constraints to have a
```

Comment:
Sorry for nit-picking: I [don't believe ANSI SQL allows CONSTRAINT foo NULL](https://jakewheat.github.io/sql-overview/sql-2011-foundation-grammar.html#column-constraint-definition) - it's Postgres that seems to allow it judging from its docs...

Reply:
Oh man, good point. What a nightmare. Fixed.